src/diffusers/models/activations.py

```diff
@@ -79,10 +79,10 @@ def __init__(self, dim_in: int, dim_out: int, approximate: str = "none", bias: b
         self.approximate = approximate
 
     def gelu(self, gate: torch.Tensor) -> torch.Tensor:
-        if gate.device.type != "mps":
-            return F.gelu(gate, approximate=self.approximate)
-        # mps: gelu is not implemented for float16
-        return F.gelu(gate.to(dtype=torch.float32), approximate=self.approximate).to(dtype=gate.dtype)
+        if gate.device.type == "mps" and torch.__version__ < '2.0.0':
```

Comment:
we have a is_torch_version function that you can use here https://github.com/huggingface/diffusers/blob/fa3a9100bede6d45c84a68f10cef45a7c562ac94/src/diffusers/utils/import_utils.py#L690